File: BleModule/src/main/java/com/topdon/ble/util/Logger.java
package com.topdon.ble.util;

public interface Logger {

    int TYPE_GENERAL = 0;

    int TYPE_SCAN_STATE = 1;

    int TYPE_CONNECTION_STATE = 2;

    int TYPE_CHARACTERISTIC_READ = 3;

    int TYPE_CHARACTERISTIC_CHANGED = 4;

    int TYPE_READ_REMOTE_RSSI = 5;

    int TYPE_MTU_CHANGED = 6;

    int TYPE_REQUEST_FAILED = 7;
    int TYPE_DESCRIPTOR_READ = 8;
    int TYPE_NOTIFICATION_CHANGED = 9;
    int TYPE_INDICATION_CHANGED = 10;
    int TYPE_CHARACTERISTIC_WRITE = 11;
    int TYPE_PHY_CHANGE = 12;

    void log(int priority, int type, String msg);

    void log(int priority, int type, String msg, Throwable th);

    boolean isEnabled();

    void setEnabled(boolean isEnabled);
}

File: ble-core/src/main/java/com/mpdc4gsr/ble/core/util/Logger.kt
package com.mpdc4gsr.ble.core.util

interface Logger {
    fun log(priority: Int, type: Int, msg: String?)

    fun log(priority: Int, type: Int, msg: String?, th: Throwable?)

    var isEnabled: Boolean

    companion object {
        const val TYPE_GENERAL: Int = 0

        const val TYPE_SCAN_STATE: Int = 1

        const val TYPE_CONNECTION_STATE: Int = 2

        const val TYPE_CHARACTERISTIC_READ: Int = 3

        const val TYPE_CHARACTERISTIC_CHANGED: Int = 4

        const val TYPE_READ_REMOTE_RSSI: Int = 5

        const val TYPE_MTU_CHANGED: Int = 6

        const val TYPE_REQUEST_FAILED: Int = 7
        const val TYPE_DESCRIPTOR_READ: Int = 8
        const val TYPE_NOTIFICATION_CHANGED: Int = 9
        const val TYPE_INDICATION_CHANGED: Int = 10
        const val TYPE_CHARACTERISTIC_WRITE: Int = 11
        const val TYPE_PHY_CHANGE: Int = 12
    }
}
